src/ActionBmwCodeCell.test.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2023 Inclusive Design Research Centre, OCAD University
+ * All rights reserved.
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/inclusive-design/adaptive-palette/blob/main/LICENSE
+ */
+
+import { render, screen } from "@testing-library/preact";
+import "@testing-library/jest-dom";
+import { html } from "htm/preact";
+
+import { initAdaptivePaletteGlobals } from "./GlobalData";
+import { ActionBmwCodeCell } from "./ActionBmwCodeCell";
+
+test("The ActionBmwCodeCell is rendered correctly", async () => {
+
+  await initAdaptivePaletteGlobals();
+
+  const TEST_CELL_ID = "uuid-of-some-kind";
+  const testCell = {
+    options: {
+      "label": "Bliss Language",
+      "rowStart": "3",
+      "rowSpan": "2",
+      "columnStart": "2",
+      "columnSpan": "1",
+      "bciAvId": [ 12335, "/", 8499 ]   // VERB+EN
+    }
+  };
+  render(html`
+    <${ActionBmwCodeCell}
+      id="${TEST_CELL_ID}"
+      options=${testCell.options}
+    />`
+  );
+
+  // Check the rendered cell
+  const button = await screen.findByRole("button", {name: testCell.options.label});
+
+  // Check that the ActionBmwCodeCell/button is rendered and has the correct
+  // attributes and text.
+  expect(button).toBeVisible();
+  expect(button).toBeValid();
+  expect(button.id).toBe(TEST_CELL_ID);
+  expect(button.getAttribute("class")).toBe("actionBmwCodeCell");
+  expect(button.textContent).toBe(testCell.options.label);
+
+  // Check the grid cell styles.
+  expect(button.style["grid-column"]).toBe("2 / span 1");
+  expect(button.style["grid-row"]).toBe("3 / span 2");
+
+  // Check disabled state (should be enabled)
```

Comment:
As the `ActionBmwCodeCell` component doesn't handle the disabled state at this point, the check on this attribute probably should be removed. It can be added when the disabled state is handled.

Reply:
I have mixed feelings about removing this.  The test specifically checks if the `button` element is enabled since, as you say, there is no disabled state at this point.  If this test fails, something is very wrong.  When the disabled state is added, then the test will fail, acting as a reminder that there needs to be an additional test for cases where the disabled state is set.

Perhaps we should create an issue about handling the disabled state so we don't forget?

Reply:
Good point. I understand your concern and agree. Let's leave the check of the disabled state in there then.